source/NVDAObjects/JAB/__init__.py

```diff
@@ -650,8 +650,6 @@ def _get_tableID(self):
 
 class TableCell(JAB):
 
-	role=controlTypes.Role.TABLECELL
-
 	def _get_table(self):

```

Comment:
I don't think changing the role of `JAB.TableCell` from `controlTypes.Role.TABLECELL` is the correct approach here. Is there no checkbox JAB control?

Reply:
NVDA has no JAB overlay class for checkbox. What such a class would contain I don't know as checkboxes work fine without, so what would such a class add? Also seeing how overlay classes get applied for JAB, we already have an issue that elif is used, so if a combobox is a direct child of a table then the Combobox overlay class is applied but not the table cell overlay class. If checkboxes were also to be a overlay class, should it be applied in the same way through an elif?
The problem is that in the Java Swing toolkit there is no such thing as a table cell control or table cell role, so to force table cell on a control which is the direct child of a table means you mask the correct role of the control.
Removal of the table cell role seems to work fine, NVDA is still aware of the table cell stuff as the overlay class is still applied (eg. it still reports column/row position).

Reply:
I agree with this approach in principle.
Java Access Bridge itself has no tableCell role. Authors place controls of what ever type on the grid as cells. I think the original reason I hardcoded role to tableCell many years back was that the tables I had seen at the time all used panel or label as the role for the cells, and it sounded a little odd. At the time I was unaware though that controls such as checkboxes could be used as cells.
Therefore, I think it is right to remove this role override and use the existing (more correct) role underneath.
However, at very least, I would suggest changing this to a proprty getter, e.g.
```
def _get_role(self):
```
Calling super and at very least checking if the role is Role.UNKNOWN and returning role.TABLECELL in that case. As it seems that our JAB gainFocus event handler specifically ignores NVDAObjects with a Role of UNKNOWN. So removing role tableCell may then start causing NVDA to ignore some gainFocus events where the author had not set the role on the tableCell.
Also, you may wish to suggest checking for a small subset of roles, such as label and panel, and instead returning tableCell. Roles that clearly have been chosen purely because there is not a tableCell role. Thus ensuring the original behaviour I saw does not appear again. Though I'd also be happy if you did not do this, keeping it more accurate to what the Author chose.